### PR TITLE
Add documentation for using mux to serve a SPA

### LINKED ISSUE #464: Serve React Router
Go Version: 1.12.

Mux-Version: latest (not important)

Iam able to serve a react app which I created with "create-react-app", but how I serve an app which is created with "create-react-app" and has react router in it? Is this possible?

### PR DESCRIPTION
**Pull request by fharding1**
Fixes #464

**Summary of Changes**

1. Adds documentation to the README.md for using mux to serve API endpoints alongside a SPA.

**Testing**

I ran create-react-app, made a prod build, copied the build directory, and tested this out. Seems to work great.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -212,6 +213,77 @@ func main() {
...
+		return
+	} else if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	http.FileServer(http.Dir(h.staticPath)).ServeHTTP(w, r)
+}
+
+func main() {
```
Comment (elithrar) [inline README.md]: `func main()` LGTM

Comment (fharding1) [reply README.md]: Woah how did that happen. Swore I compiled this before PRing :thinking:

File: README.md [hunk 2]
```diff
@@ -212,6 +213,77 @@ func main() {
...
+	"log"
+	"net/http"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/gorilla/mux"
+)
+
+type spaHandler struct {
```
[the remark this answers is not in the document]
Comment (fharding1) [reply README.md]: Hmm. I can see how a `http.HandlerFunc` might be easier for a beginner than a type implementing `http.Handler`, however I think the overhead of the closure might outweigh that. Overall I think both are about as easy for a beginner to understand (also open to feedback of course :smile:).

[the remark this answers is not in the document]
Comment (elithrar) [reply README.md]: So, thinking on this: should we add a “ServeStatic” or similar handler as a built-in? This has come up a few times.

File: README.md [hunk 3]
```diff
@@ -212,6 +213,77 @@ func main() {
...
+}
+
+func (h spaHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	path, err := filepath.Abs(r.URL.Path)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	path = filepath.Join(h.staticPath, path)
```
Comment (elithrar) [inline README.md]: My general feedback here is that we (over) explain this function body.

**Comment by fharding1** [comment]
I think implementing a built-in handler in `gorilla/mux` for the few people trying to use the wrong tool for the job (`gorilla/mux` to serve an SPA) wouldn't be worth adding more stuff to the API. Instead, I think it makes sense to leave this as docs so that people will at least stop opening issues asking how to serve an SPA with mux.

File: README.md (hunk 3 above)
Comment (fharding1) [reply README.md]: Agreed, it should be more clear how this works.

**Comment by fharding1** [comment]
Let me know if you think it still makes sense to add a built-in handler with the docs improvements and I'll implement that :+1:

**Comment by elithrar** [comment]
Let's leave it as-is for now, @fharding1 - we can revisit later. `http.FileServer` continues to trip users up, so wary of that.

**Comment by elithrar** [comment]
(Thanks for working on this, too!)

**Comment by fharding1** [comment]
Np, thank you!

**Pull request merged**